src/lib/operators/delete.cpp

```diff
@@ -89,13 +90,13 @@ std::shared_ptr<const Table> Delete::_on_execute(std::shared_ptr<TransactionCont
 void Delete::_on_commit_records(const CommitID cid) {
   for (ChunkID referencing_chunk_id{0}; referencing_chunk_id < _referencing_table->chunk_count();
        ++referencing_chunk_id) {
-    const auto referencing_chunk = _referencing_table->get_chunk(referencing_chunk_id);
+    const auto& referencing_chunk = _referencing_table->get_chunk(referencing_chunk_id);
```

Comment:
I just realized that `const auto&` is wrong here. Imagine `get_chunk` returned a reference. With this change, we would not copy the `shared_ptr` anymore, so we have no guarantee that the object is not suddenly deleted.